packages/now-cli/src/util/projects/link.ts

```diff
@@ -102,22 +102,28 @@ export async function linkFolderToProject(
     }
   );
 
-  // update .nowignore
+  // update .gitignore
+  let isGitIgnoreUpdated = false;
   try {
     const gitIgnorePath = join(path, '.gitignore');
-    const gitIgnore = (await readFile(gitIgnorePath)).toString();
-    if (gitIgnore.split('\n').indexOf('.now') < 0) {
-      await writeFile(gitIgnorePath, gitIgnore + '\n.now');
+
+    const gitIgnore = await readFile(gitIgnorePath)
+      .then(buf => buf.toString())
+      .catch(() => null);
+
+    if (!gitIgnore || !gitIgnore.split('\n').includes('.now')) {
```

Comment:
I think we need to be careful when an existing `.gitignore` file is found.

Some people will do a whitelist/allowlist pattern where they ignore everything and then selectively add files such as:

```gitignore
*
!src
!src/*
index.js
```

So maybe something like `!gitIgnoreLines.includes('.now') && !gitIgnoreLines.includes('*')`.

Reply:
Actually, I realize that this won't be a problem because we are not going to prefix with a `!` 🤦‍♂